src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -108,29 +108,29 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
         }
 
         try {
-            if (null !== $dateTimeFormat) {
-                $object = \DateTime::class === $type ? \DateTime::createFromFormat($dateTimeFormat, $data, $timezone) : \DateTimeImmutable::createFromFormat($dateTimeFormat, $data, $timezone);
+            if (\DateTimeInterface::class === $type) {
+                $type = \DateTimeImmutable::class;
```

Comment:
using  `\DatetimeImmutable` instead of `$type` everywhere would read better imho

Reply:
this would break when $type is "DateTime" or a child class of "DateTime(Immutable)"